core/templates/dev/head/app.js

```diff
@@ -65,8 +65,10 @@ oppia.config(['$interpolateProvider', '$httpProvider', '$animateProvider',
         },
         responseError: function(response) {
           $log.error(response.data);
-          warningsData.addWarning(
-            response.data.error || 'Error communicating with server.');
+          if (response.data) {
+            warningsData.addWarning(
+              response.data.error || 'Error communicating with server.');
+          }
```

Comment:
I think we should still show a warning in the case when there's no response.data.

Maybe
var errorMessage = (response.data && response.data.error) ? response.data.error : 'Error communicating with server.'

Having said that, what prompted this change? Maybe we should instead just fix the backend to always return a 'data' payload.